src/main/java/co/cask/hydrator/plugin/PartitionedFileSetSinkConfig.java

```diff
@@ -140,7 +140,10 @@ protected Partitioning getPartitioningWithoutMacro(Schema inputSchema) {
   /**
    * Validate plugin configuration at deployment stage.
    */
-  protected void validate(Schema inputSchema) {
+  protected void validate(@Nullable Schema inputSchema) {
+    if (inputSchema == null) {
+      return;
```

Comment:
if we allow `inputSchema` to be null, `getPartitioningWithoutMacro` method should also be changed, where it assumes `inputSchema` is not null